### PR TITLE
Convert Extension tab to use Compose

### PR DESCRIPTION
**Pull request by ghostbear**
- Makes Extension tab use Compose
- Untrusted extensions now have a different icon

image

### CODE REVIEW

File: app/src/main/java/eu/kanade/tachiyomi/ui/browse/extension/ExtensionPresenter.kt [hunk 1]
```diff
@@ -2,144 +2,146 @@ package eu.kanade.tachiyomi.ui.browse.extension
...
-        val items = mutableListOf<ExtensionItem>()
+        val extensionMapper: (Map<String, InstallStep>) -> ((Extension) -> ExtensionUiModel) = { map ->
+            {
+                ExtensionUiModel.Item(it, map[it.pkgName] ?: InstallStep.Idle)
+            }
+        }
+        val queryFilter: (String) -> ((Extension) -> Boolean) = { query ->
+            filter@{ extension ->
+                if (query.isEmpty()) return@filter true
+                query.split(",").any { query ->
```
Comment (FourTOne5) [inline app/src/main/java/eu/kanade/tachiyomi/ui/browse/extension/ExtensionPresenter.kt:61]: Trim the query?

File: app/src/main/java/eu/kanade/tachiyomi/ui/browse/extension/ExtensionViewUtils.kt [hunk 2]
```diff
@@ -12,3 +17,15 @@ fun Extension.getApplicationIcon(context: Context): Drawable? {
         null
     }
 }
+
+@Composable
+fun Extension.getIcon(): ImageBitmap? {
+    val context = LocalContext.current
+    return try {
+        context.packageManager.getApplicationIcon(pkgName)
```
Comment (jobobby04) [inline app/src/main/java/eu/kanade/tachiyomi/ui/browse/extension/ExtensionViewUtils.kt:25]: Won't this happen on the UI thread?

File: app/src/main/java/eu/kanade/tachiyomi/ui/browse/extension/ExtensionPresenter.kt [hunk 3]
```diff
@@ -2,144 +2,150 @@ package eu.kanade.tachiyomi.ui.browse.extension
...
-
-        val items = mutableListOf<ExtensionItem>()
+        val extensionMapper: (Map<String, InstallStep>) -> ((Extension) -> ExtensionUiModel) = { map ->
+            {
+                ExtensionUiModel.Item(it, map[it.pkgName] ?: InstallStep.Idle)
+            }
+        }
+        val queryFilter: (String) -> ((Extension) -> Boolean) = {
+            filter@{ extension ->
+                val query = it.trim()
```
Comment (FourTOne5) [inline app/src/main/java/eu/kanade/tachiyomi/ui/browse/extension/ExtensionPresenter.kt:60]: I meant

**Comment by Soitora** [comment]
Test comment

File: app/src/main/java/eu/kanade/presentation/extension/ExtensionScreen.kt [hunk 4]
```diff
@@ -0,0 +1,417 @@
...
+) {
+    val (extension, installStep) = item
+    BaseBrowseItem(
+        modifier = modifier
+            .combinedClickable(
+                onClick = { onClickItem(extension) },
+                onLongClick = { onLongClickItem(extension) },
+            ),
+        onClickItem = { onClickItem(extension) },
+        onLongClickItem = { onLongClickItem(extension) },
```
[the remark this answers is not in the document]
Comment (ghostbear) [reply app/src/main/java/eu/kanade/presentation/extension/ExtensionScreen.kt:234]: Yeah, a bit I was thinking of letting the Compose screen interact more with the presenter and not have the controller do all the work

**Pull request merged**